plugins/discourse-local-dates/assets/javascripts/lib/discourse-markdown/discourse-local-dates.js

```diff
@@ -162,7 +162,9 @@ function addLocalRange(buffer, matches, state) {
     addSingleLocalDate(buffer, state, config);
   }
   if (config.range) {
-    closeBuffer(buffer, state, "→");
+    const token = new state.Token("text", "", 0);
```

Comment:
This fixes the nesting issue but directly adding a text token with a nesting of 0, instead of calling `closeBuffer` which adds two tokens with nesting 0 and -1 (missing the nesting = 1).